frontend/src/components/Calender.js

```diff
@@ -0,0 +1,62 @@
+import React, { useEffect, useState } from 'react';
+import { Box, Typography, TextField } from '@mui/material';
+import { DatePicker } from '@mui/x-date-pickers';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import { AdapterDateFns } from '@mui/x-date-pickers/AdapterDateFns';
+import { apiCallWithAuth } from 'utils/authAPI';
+import { format, isAfter, isSameDay, parseISO } from 'date-fns';
+
+const ArticleCalendar = ({ onDateChange }) => {
+  const [articleDates, setArticleDates] = useState([]);
+  const [selectedDate, setSelectedDate] = useState(null);
+
+  useEffect(() => {
+    const fetchDates = async () => {
+      try {
+        const response = await apiCallWithAuth('/api/article-dates/');
+        if (response.success) {
+          const dates = response.data.dates.map(date => parseISO(date));
+          setArticleDates(dates);
+        } else {
+          console.error('Error fetching article dates:', response.errors);
+        }
+      } catch (error) {
+        console.error('Error fetching article dates:', error);
+      }
+    };
+
+    fetchDates();
+  }, []);
+
+  useEffect(() => {
+    if (selectedDate) {
+    // Notify the parent component of the selected date change i.e (Dashboard.js)
+      onDateChange(selectedDate);
+    }
+  }, [selectedDate, onDateChange]);
```

Comment:
Do you think we need this useEffect? We are already sending the date to the parent in DatePicker onChange.

Reply:
you are right, we don't need this.